src/Esprima/Scanner.RegExpParser.cs

```diff
@@ -185,7 +185,6 @@ public RegExpParseResult Parse()
             }
 
             Debug.Assert(conversionError is null);
-            capturingGroups.TrimExcess();
```

Comment:
IMO we'd better trim the collected `capturingGroups` one way or the other because it's exposed in the public API (see `RegExpParseResult`).